rts/System/Threading/ThreadPool.cpp

```diff
@@ -108,19 +109,26 @@ static int GetConfigNumWorkers() {
 }
 
 static int GetDefaultNumWorkers() {
-	const int maxNumThreads = GetMaxThreads(); // min(MAX_THREADS, logicalCpus)
-	const int cfgNumWorkers = GetConfigNumWorkers();
+        const int maxNumThreads = GetMaxThreads(); // min(MAX_THREADS, logicalCpus)
+        const int cfgNumWorkers = GetConfigNumWorkers();
 
-	if (cfgNumWorkers < 0) {
-		return Threading::GetPerformanceCpuCores();
-	}
+        if (cfgNumWorkers >= 0) {
+                if (cfgNumWorkers > maxNumThreads) {
+                        LOG_L(L_WARNING, "[ThreadPool::%s] workers set to %i, but there are just %i hardware threads!", __func__, cfgNumWorkers, maxNumThreads);
+                        return maxNumThreads;
+                }
 
-	if (cfgNumWorkers > maxNumThreads) {
-		LOG_L(L_WARNING, "[ThreadPool::%s] workers set to %i, but there are just %i hardware threads!", __func__, cfgNumWorkers, maxNumThreads);
-		return maxNumThreads;
-	}
+                return cfgNumWorkers;
+        }
+
+        // automatic worker count
+        const unsigned int hwCores = std::thread::hardware_concurrency();
+        const int perfCores = Threading::GetPerformanceCpuCores();
 
-	return cfgNumWorkers;
+        if (cfgNumWorkers == -1 && perfCores > 0 && hwCores > static_cast<unsigned int>(perfCores))
```

Comment:
`hwCores > static_cast<unsigned int>(perfCores)` is supposed to be always true, what did you mean to check here?

Reply:
We made intentional decision to use only one logical core in a HT pair of cores as usually this yields better overall performance.
Have you observed opposite results? Can you share the testing method?